lib/puppet/type/file/source.rb

```diff
@@ -273,18 +261,18 @@ def copy_source_value(metadata_method)
       end
     end
 
-    def each_chunk_from
-      if Puppet[:default_file_terminus] == :file_server
-        yield content
+    def each_chunk_from(&block)
+      if Puppet[:default_file_terminus] == :file_server && scheme == 'puppet' && (uri.host.nil? || uri.host.empty?)
+        chunk_file_from_disk(metadata.path, &block)
```

Comment:
What's the difference between `metadata.source` and `metadata.path`? I'm not seeing anything that outlines that change

Reply:
Yeah it's confusing, the metadata object is lacking some documentation. But `metadata.source` is the value from the manifest that says "where to find the file". It can be an absolute path or puppet/http/https URL. `metadata.path` is the path where the resolved file was found based on environment, modulepath, etc.

When using the in-process file server, we can copy the file directly from `metadata.path`. But when using the remote file server, `metadata.path` is the path on the puppetserver where the file was found.

Reply:
Thank you for explaining that! This makes a lot more sense to me now